cpp/src/copying/contiguous_split.cu

```diff
@@ -0,0 +1,207 @@
+#include <cudf/cudf.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/copying.hpp>
+#include <cudf/detail/utilities/cuda.cuh>
+
+namespace cudf {
+
+namespace experimental {
+
+namespace detail {
+
+namespace {
+
+template <size_type block_size, typename T, bool has_validity>
+__launch_bounds__(block_size)
+__global__
+void copy_in_place_kernel( column_device_view const in,
```

Comment:
So I spent some time thinking about this kernel, and I think we can avoid it. 

The data portion of the columns is easy, that's just memcpys. The tricky part is the bitmasks because you can't memcpy at the granularity of individual bits. Luckily, there is already a kernel that can handle offset bitmask copies: https://github.com/rapidsai/cudf/blob/branch-0.12/cpp/src/bitmask/null_mask.cu#L188

I think this kernel combined with simple memcpys could eliminate the need for this kernel (and therefore eliminate the need for creating `column_device_view`s and the issue with `null_count()).

Reply:
Ok, so I gave this a try, and while it works, it is considerably slower in the case where we actually have validity. 

```
     cudaMemcpy(data, in.data<T>(), data_size, cudaMemcpyDeviceToDevice);
      if(in.nullable()){           
         _copy_bitmask(validity, in.null_mask(), in.offset(), in.offset() + in.size(), (cudaStream_t)0);
      }
```
I made a local copy of _copy_bitmask that takes destination bitmask as an input (instead of allocating itself).

Here's the key timings (timings done using googlebenchmarks):

(Existing PR method)
6Gb, 512 cols, 256 splits, no validity :  364 ms  (avg over 20 iterations)
6Gb, 512 cols, 256 splits, validity      :  370 ms  (avg over 20 iterations)

(New method)
6Gb, 512 cols, 256 splits, no validity :  438 ms  (avg over 20 iterations)
6Gb, 512 cols, 256 splits, validity      :  767 ms  (avg over 20 iterations)


It makes sense if you think about it.  With the PR as is, we're doing 1 kernel call per copy, regardless of the presence of validity.   With this suggested method, we're doing 2 (I'm assuming cudaMemcpy() ends up effectively a kernel).  Benchmarks have always indicated that we're bound by total # of kernel calls, so doubling them exaggerates this.